src/west/manifest.py

```diff
@@ -87,11 +87,12 @@
 _logger = logging.getLogger(__name__)
 
 # Type for the submodule value passed through the manifest file.
-class Submodule(NamedTuple):
+class Submodule:
     '''Represents a Git submodule within a project.'''
 
-    name: str
-    path: str
+    def __init__(self, path, name=None):
```

Comment:
I think it would be nicer to keep this a NamedTuple. There is support for default arguments with this type:

https://docs.python.org/3/library/typing.html#typing.NamedTuple

Moving to a class like this without type annotations sacrifices checking that mypy  can do for us. Having runtime type checking is essential, but it's also nice to have the type information available during static analysis.